core/invoker/src/main/scala/whisk/core/containerpool/ContainerPool.scala

```diff
@@ -192,13 +192,14 @@ object ContainerPool {
      * takes away one of their containers iff the namespace placing the new job is
      * not already the most consuming one.
      *
-     * @param namespace the namespace that wants to get a container
+     * @param action the action that wants to get a container
      * @param pool a map of all containers in the pool
      * @return a container to be removed iff found
      */
-    def remove[A](invocationNamespace: EntityName, pool: Map[A, WorkerData]): Option[A] = {
+    def remove[A](action: ExecutableWhiskAction, pool: Map[A, WorkerData]): Option[A] = {
```

Comment:
containers are specific to a subject+action; we don't reuse a container to run an action for more than one subject since we assume the container is tainted (ok to self taint but shall not taint another user or action).